### PR TITLE
Handle trivial CSV files / issue #222

### PR DESCRIPTION
**Pull request by tilo**
[issue #222](https://github.com/tilo/smarter_csv/issues/222)

For trivial CSV files with only one column, don't raise `NoColSepDetected`.

### CODE REVIEW

File: lib/smarter_csv.rb [hunk 1]
```diff
@@ -393,15 +393,28 @@ def only_or_except_limit_execution(options, option_name, key)
...
+      has_header = options[:headers_in_file]
+      candidates = Hash.new(0)
+      count = has_header ? 1 : 5
+      count.times do
+        line = readline_with_counts(filehandle, options)
+        delimiters.each do |d|
+          candidates[d] += line.scan(d).count
+        end
+      rescue EOFError # short files
+        break
```
Comment (dpep) [inline lib/smarter_csv.rb:408]: do you need to rewind before break?  (maybe an `ensure` block?)  is there a test that covers this corner case?

File: spec/smarter_csv/simple_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
```
Comment (dpep) [inline spec/smarter_csv/simple_spec.rb:3]: put this into `.rspec` [instead](https://github.com/rspec/rspec-core/blob/main/README.md#store-command-line-options-rspec)

```
> cat .rspec 
--require spec_helper
```

File: lib/smarter_csv.rb [hunk 3]
```diff
@@ -393,15 +393,28 @@ def only_or_except_limit_execution(options, option_name, key)
...
+      end
+      rewind(filehandle)
 
-      raise SmarterCSV::NoColSepDetected if candidates.values.max == 0
+      if candidates.values.max == 0
+        # if the header only contains
+        return ',' if line =~ /^\w+$/
```
Comment (dpep) [inline lib/smarter_csv.rb:414]: given the prevalence of UTF-8, why the special case for ascii only?

**Review by dpep — COMMENTED** [review]
seems like a step in the right direction ;)
